google/cloud/storage/internal/time_fetcher.h

```diff
@@ -0,0 +1,42 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_STORAGE_INTERNAL_TIME_FETCHER_H_
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_STORAGE_INTERNAL_TIME_FETCHER_H_
+
+#include "google/cloud/storage/version.h"
+#include <ctime>
+
+namespace google {
+namespace cloud {
+namespace storage {
+inline namespace STORAGE_CLIENT_NS {
+namespace internal {
+/**
+ * Helper to fetch current time. Allows for subclassing to fake current time.
+ */
+struct TimeFetcher {
+ public:
+  virtual long int GetSecondsSinceEpoch() {
+    return static_cast<long int>(std::time(nullptr));
```

Comment:
Elsewhere we have been using `std::chrono::system_clock` for this type of thing.  Consider this instead, define `ServiceAccountCredentials` as:

```C++
template <typename HttpRequestType = CurlRequest,
          typename ClockType = std::chrono::system_clock>
class ServiceAccountCredentials : public storage::Credentials {
```

and then define:

```C++
class MockClock : public std::chrono::system_clock {
  static std::chrono::system_clock::time_point now() { return hard_coded_value; }
};
```

Reply:
Done.